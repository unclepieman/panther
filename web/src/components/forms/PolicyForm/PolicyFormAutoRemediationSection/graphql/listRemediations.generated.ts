/**
 * Panther is a Cloud-Native SIEM for the Modern Security Team.
 * Copyright (C) 2020 Panther Labs Inc
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

import * as Types from '../../../../../../__generated__/schema';

import { GraphQLError } from 'graphql';
import gql from 'graphql-tag';
import * as ApolloReactCommon from '@apollo/client';
import * as ApolloReactHooks from '@apollo/client';

export type ListRemediationsVariables = {};

export type ListRemediations = Pick<Types.Query, 'remediations'>;

export const ListRemediationsDocument = gql`
  query ListRemediations {
    remediations
  }
`;

/**
 * __useListRemediations__
 *
 * To run a query within a React component, call `useListRemediations` and pass it any options that fit your needs.
 * When your component renders, `useListRemediations` returns an object from Apollo Client that contains loading, error, and data properties
 * you can use to render your UI.
 *
 * @param baseOptions options that will be passed into the query, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options;
 *
 * @example
 * const { data, loading, error } = useListRemediations({
 *   variables: {
 *   },
 * });
 */
export function useListRemediations(
  baseOptions?: ApolloReactHooks.QueryHookOptions<ListRemediations, ListRemediationsVariables>
) {
  return ApolloReactHooks.useQuery<ListRemediations, ListRemediationsVariables>(
    ListRemediationsDocument,
    baseOptions
  );
}
export function useListRemediationsLazyQuery(
  baseOptions?: ApolloReactHooks.LazyQueryHookOptions<ListRemediations, ListRemediationsVariables>
) {
  return ApolloReactHooks.useLazyQuery<ListRemediations, ListRemediationsVariables>(
    ListRemediationsDocument,
    baseOptions
  );
}
export type ListRemediationsHookResult = ReturnType<typeof useListRemediations>;
export type ListRemediationsLazyQueryHookResult = ReturnType<typeof useListRemediationsLazyQuery>;
export type ListRemediationsQueryResult = ApolloReactCommon.QueryResult<
  ListRemediations,
  ListRemediationsVariables
>;
export function mockListRemediations({
  data,
  variables,
  errors,
}: {
  data: ListRemediations;
  variables?: ListRemediationsVariables;
  errors?: GraphQLError[];
}) {
  return {
    request: { query: ListRemediationsDocument, variables },
    result: { data, errors },
  };
}
