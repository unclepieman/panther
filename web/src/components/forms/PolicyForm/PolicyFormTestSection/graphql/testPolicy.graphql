mutation TestPolicy($input: TestPolicyInput!) {
  testPolicy(input: $input) {
    results {
      id
      name
      passed
      error {
        message
      }
      functions {
        policyFunction {
          ...TestFunctionResult
        }
      }
    }
  }
}
