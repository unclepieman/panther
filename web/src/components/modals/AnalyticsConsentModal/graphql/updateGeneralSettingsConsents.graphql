mutation UpdateGeneralSettingsConsents($input: UpdateGeneralSettingsInput!) {
    updateGeneralSettings(input: $input) {
        email
        errorReportingConsent
        analyticsConsent
    }
}