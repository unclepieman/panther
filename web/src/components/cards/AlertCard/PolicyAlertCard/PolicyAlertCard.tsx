/**
 * Panther is a Cloud-Native SIEM for the Modern Security Team.
 * Copyright (C) 2020 Panther Labs Inc
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

import GenericItemCard from 'Components/GenericItemCard';
import { Flex, Icon, Link, Text, Box, Divider, Grid } from 'pouncejs';
import { AlertSummaryPolicyInfo } from 'Generated/schema';
import { Link as RRLink } from 'react-router-dom';
import SeverityBadge from 'Components/badges/SeverityBadge';
import React from 'react';
import urls from 'Source/urls';
import RelatedDestinations from 'Components/RelatedDestinations';
import BulletedValueList from 'Components/BulletedValueList';
import { AlertSummaryFull } from 'Source/graphql/fragments/AlertSummaryFull.generated';
import { formatDatetime } from 'Helpers/utils';
import { useListComplianceSourceNames } from 'Source/graphql/queries';
import useAlertDestinations from 'Hooks/useAlertDestinations';
import useAlertDestinationsDeliverySuccess from 'Hooks/useAlertDestinationsDeliverySuccess';
import { SelectCheckbox } from 'Components/utils/SelectContext';
import UpdateAlertDropdown from 'Components/dropdowns/UpdateAlertDropdown';
import FlatBadge from 'Components/badges/FlatBadge';

export interface PolicyAlertCardProps {
  alert: AlertSummaryFull;
  hidePolicyButton?: boolean;
  selectionEnabled?: boolean;
}

const PolicyAlertCard: React.FC<PolicyAlertCardProps> = ({
  alert,
  hidePolicyButton = false,
  selectionEnabled = false,
}) => {
  const { data: complianceSourceData } = useListComplianceSourceNames({ errorPolicy: 'ignore' });
  const { alertDestinations, loading: loadingDestinations } = useAlertDestinations({ alert });
  const { allDestinationDeliveredSuccessfully, loading } = useAlertDestinationsDeliverySuccess({
    alert,
  });

  const detectionData = alert.detection as AlertSummaryPolicyInfo;
  const source = complianceSourceData?.listComplianceIntegrations?.find(
    s => s.integrationId === detectionData.policySourceId
  );
  return (
    <GenericItemCard>
      <Flex align="start" pr={2}>
        {selectionEnabled && (
          <Box transform="translate3d(0,-8px,0)">
            <SelectCheckbox selectionItem={alert.alertId} />
          </Box>
        )}
      </Flex>
      <GenericItemCard.Body>
        <GenericItemCard.Header>
          <GenericItemCard.Heading>
            <Link
              as={RRLink}
              aria-label="Link to Alert"
              to={urls.logAnalysis.alerts.details(alert.alertId)}
            >
              {alert.title}
            </Link>
          </GenericItemCard.Heading>
          {source && (
            <GenericItemCard.HeadingValue
              value={source.integrationLabel}
              label="Source"
              labelFirst
              withDivider
            />
          )}
          <GenericItemCard.HeadingValue
            aria-label={`Creation time for ${alert.alertId}`}
            value={formatDatetime(alert.creationTime)}
            label="Created"
            labelFirst
          />
        </GenericItemCard.Header>
        <Box mr="auto">
          <FlatBadge color="cyan-400">POLICY FAIL</FlatBadge>
        </Box>
        <Grid gap={2} templateColumns="3fr 4fr 3fr">
          <Box>
            {!hidePolicyButton && (
              <GenericItemCard.Value
                label="Policy"
                value={
                  <Flex align="center" spacing={2}>
                    <Text maxWidth={250} truncated alignItems="center" as="span">
                      {detectionData.policyId}
                    </Text>
                    <GenericItemCard.Link
                      aria-label={`Link to policy ${detectionData.policyId}`}
                      to={urls.compliance.policies.details(detectionData.policyId)}
                    />
                  </Flex>
                }
              />
            )}
          </Box>
          <GenericItemCard.Value
            label="Resource Types"
            value={<BulletedValueList values={detectionData.resourceTypes} limit={2} />}
          />
          <Flex align="flex-end">
            <Flex spacing={2} align="center" width="100%" justify="flex-end">
              <RelatedDestinations destinations={alertDestinations} loading={loadingDestinations} />
              <Divider mx={0} alignSelf="stretch" orientation="vertical"></Divider>
              <SeverityBadge severity={alert.severity} />
              <UpdateAlertDropdown alert={alert} />
            </Flex>
          </Flex>
        </Grid>
        {!loading && !allDestinationDeliveredSuccessfully && (
          <Flex
            as="section"
            align="center"
            spacing={1}
            mt={2}
            aria-label="Destination delivery failure"
            fontStyle="italic"
            color="red-100"
            fontSize="small"
          >
            <Icon type="alert-circle-filled" size="medium" />
            <Text>
              There was an issue with the delivery of this alert to a selected destination.
            </Text>
            <RRLink to={urls.logAnalysis.alerts.details(alert.alertId)}>
              <Text textDecoration="underline">See details</Text>
            </RRLink>
          </Flex>
        )}
      </GenericItemCard.Body>
    </GenericItemCard>
  );
};

export default React.memo(PolicyAlertCard);
