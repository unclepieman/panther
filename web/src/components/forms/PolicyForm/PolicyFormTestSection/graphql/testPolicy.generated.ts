/**
 * Panther is a Cloud-Native SIEM for the Modern Security Team.
 * Copyright (C) 2020 Panther Labs Inc
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

import * as Types from '../../../../../../__generated__/schema';

import { TestFunctionResult } from '../../../../../graphql/fragments/TestFunctionResult.generated';
import { GraphQLError } from 'graphql';
import gql from 'graphql-tag';
import * as ApolloReactCommon from '@apollo/client';
import * as ApolloReactHooks from '@apollo/client';

export type TestPolicyVariables = {
  input: Types.TestPolicyInput;
};

export type TestPolicy = {
  testPolicy: {
    results: Array<
      Pick<Types.TestPolicyRecord, 'id' | 'name' | 'passed'> & {
        error?: Types.Maybe<Pick<Types.Error, 'message'>>;
        functions: { policyFunction: TestFunctionResult };
      }
    >;
  };
};

export const TestPolicyDocument = gql`
  mutation TestPolicy($input: TestPolicyInput!) {
    testPolicy(input: $input) {
      results {
        id
        name
        passed
        error {
          message
        }
        functions {
          policyFunction {
            ...TestFunctionResult
          }
        }
      }
    }
  }
  ${TestFunctionResult}
`;
export type TestPolicyMutationFn = ApolloReactCommon.MutationFunction<
  TestPolicy,
  TestPolicyVariables
>;

/**
 * __useTestPolicy__
 *
 * To run a mutation, you first call `useTestPolicy` within a React component and pass it any options that fit your needs.
 * When your component renders, `useTestPolicy` returns a tuple that includes:
 * - A mutate function that you can call at any time to execute the mutation
 * - An object with fields that represent the current status of the mutation's execution
 *
 * @param baseOptions options that will be passed into the mutation, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options-2;
 *
 * @example
 * const [testPolicy, { data, loading, error }] = useTestPolicy({
 *   variables: {
 *      input: // value for 'input'
 *   },
 * });
 */
export function useTestPolicy(
  baseOptions?: ApolloReactHooks.MutationHookOptions<TestPolicy, TestPolicyVariables>
) {
  return ApolloReactHooks.useMutation<TestPolicy, TestPolicyVariables>(
    TestPolicyDocument,
    baseOptions
  );
}
export type TestPolicyHookResult = ReturnType<typeof useTestPolicy>;
export type TestPolicyMutationResult = ApolloReactCommon.MutationResult<TestPolicy>;
export type TestPolicyMutationOptions = ApolloReactCommon.BaseMutationOptions<
  TestPolicy,
  TestPolicyVariables
>;
export function mockTestPolicy({
  data,
  variables,
  errors,
}: {
  data: TestPolicy;
  variables?: TestPolicyVariables;
  errors?: GraphQLError[];
}) {
  return {
    request: { query: TestPolicyDocument, variables },
    result: { data, errors },
  };
}
