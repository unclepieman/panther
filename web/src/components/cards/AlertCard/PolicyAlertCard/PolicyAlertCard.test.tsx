/**
 * Panther is a Cloud-Native SIEM for the Modern Security Team.
 * Copyright (C) 2020 Panther Labs Inc
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

import {
  buildAlertSummary,
  buildAlertSummaryPolicyInfo,
  buildComplianceIntegration,
  buildDeliveryResponse,
  buildDestination,
  render,
  waitForElementToBeRemoved,
} from 'test-utils';
import React from 'react';
import {
  AlertStatusesEnum,
  AlertSummaryPolicyInfo,
  DestinationTypeEnum,
  SeverityEnum,
} from 'Generated/schema';
import urls from 'Source/urls';
import { mockListComplianceSourceNames, mockListDestinations } from 'Source/graphql/queries';
import PolicyAlertCard from './index';

describe('PolicyAlertCard', () => {
  it('should match snapshot', async () => {
    const alert = buildAlertSummary({ detection: buildAlertSummaryPolicyInfo() });

    const { container } = render(<PolicyAlertCard alert={alert} />);

    expect(container).toMatchSnapshot();
  });

  it('displays the correct Alert data in the card', async () => {
    const alert = buildAlertSummary({
      detection: buildAlertSummaryPolicyInfo({
        resourceTypes: ['alpha', 'beta'],
      }),
    });

    const detectionData = alert.detection as AlertSummaryPolicyInfo;

    const { getByText, getByAriaLabel } = render(<PolicyAlertCard alert={alert} />);

    expect(getByText(alert.title)).toBeInTheDocument();
    expect(getByAriaLabel(`Link to policy ${detectionData.policyId}`)).toBeInTheDocument();
    expect(getByAriaLabel(`Creation time for ${alert.alertId}`)).toBeInTheDocument();
    expect(getByText(SeverityEnum.Medium)).toBeInTheDocument();
    expect(getByText(AlertStatusesEnum.Triaged)).toBeInTheDocument();
    expect(getByAriaLabel('Change Alert Status')).toBeInTheDocument();

    detectionData.resourceTypes.forEach(resourceType => {
      expect(getByText(resourceType)).toBeInTheDocument();
    });
  });

  it('should not display link to Policy', async () => {
    const alert = buildAlertSummary({ detection: buildAlertSummaryPolicyInfo() });
    const detectionData = alert.detection as AlertSummaryPolicyInfo;

    const { queryByText, queryByAriaLabel } = render(
      <PolicyAlertCard alert={alert} hidePolicyButton />
    );

    expect(queryByText(alert.title)).toBeInTheDocument();
    expect(queryByAriaLabel(`Link to policy ${detectionData.policyId}`)).not.toBeInTheDocument();
  });

  it('should check links are valid', async () => {
    const alert = buildAlertSummary({ detection: buildAlertSummaryPolicyInfo() });
    const detectionData = alert.detection as AlertSummaryPolicyInfo;

    const { getByAriaLabel } = render(<PolicyAlertCard alert={alert} />);

    expect(getByAriaLabel('Link to Alert')).toHaveAttribute(
      'href',
      urls.logAnalysis.alerts.details(alert.alertId)
    );
    expect(getByAriaLabel(`Link to policy ${detectionData.policyId}`)).toHaveAttribute(
      'href',
      urls.compliance.policies.details(detectionData.policyId)
    );
  });

  it('should render alert destinations logos', async () => {
    const outputId = 'destination-of-alert';
    const alert = buildAlertSummary({
      detection: buildAlertSummaryPolicyInfo(),
      deliveryResponses: [buildDeliveryResponse({ outputId })],
    });
    const destination = buildDestination({ outputId, outputType: DestinationTypeEnum.Slack });
    const mocks = [mockListDestinations({ data: { destinations: [destination] } })];

    const { getByAriaLabel, getByAltText } = render(<PolicyAlertCard alert={alert} />, {
      mocks,
    });
    const loadingInterfaceElement = getByAriaLabel('Loading...');
    expect(loadingInterfaceElement).toBeInTheDocument();
    await waitForElementToBeRemoved(loadingInterfaceElement);
    expect(getByAltText(`${destination.outputType} logo`)).toBeInTheDocument();
  });

  it("should render the source's name", async () => {
    const sourceId = 'destination-of-alert';
    const alert = buildAlertSummary({
      detection: buildAlertSummaryPolicyInfo({ policySourceId: sourceId }),
    });
    const mocks = [
      mockListComplianceSourceNames({
        data: {
          listComplianceIntegrations: [
            buildComplianceIntegration({
              integrationId: sourceId,
              integrationLabel: 'fake-source-name',
            }),
          ],
        },
      }),
    ];

    const { findByText } = render(<PolicyAlertCard alert={alert} />, {
      mocks,
    });

    expect(await findByText('fake-source-name')).toBeInTheDocument();
  });

  it('should render message that destination delivery is failing', async () => {
    const outputId = 'destination-of-alert';
    const alert = buildAlertSummary({
      detection: buildAlertSummaryPolicyInfo(),
      deliveryResponses: [buildDeliveryResponse({ outputId, success: false })],
    });
    const destination = buildDestination({ outputId, outputType: DestinationTypeEnum.Slack });
    const mocks = [mockListDestinations({ data: { destinations: [destination] } })];

    const { getByAriaLabel, getByAltText } = render(<PolicyAlertCard alert={alert} />, {
      mocks,
    });
    const loadingInterfaceElement = getByAriaLabel('Loading...');
    expect(loadingInterfaceElement).toBeInTheDocument();
    await waitForElementToBeRemoved(loadingInterfaceElement);
    expect(getByAltText(`${destination.outputType} logo`)).toBeInTheDocument();
    expect(getByAriaLabel('Destination delivery failure')).toBeInTheDocument();
  });
});
