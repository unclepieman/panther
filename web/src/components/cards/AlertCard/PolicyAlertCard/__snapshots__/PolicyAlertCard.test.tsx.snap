// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PolicyAlertCard should match snapshot 1`] = `
@keyframes animation-0 {
  0% {
    -webkit-transform: rotate(0);
    -moz-transform: rotate(0);
    -ms-transform: rotate(0);
    transform: rotate(0);
  }

  100% {
    -webkit-transform: rotate(360deg);
    -moz-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

@keyframes animation-1 {
  0% {
    stroke-dashoffset: 600;
  }

  100% {
    stroke-dashoffset: 0;
  }
}

.panther-40 {
  background-color: #1d2a3f;
  border-radius: 4px;
  overflow: hidden;
}

.panther-39 {
  background-color: transparent;
  padding: 16px;
}

.panther-38 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  height: 100%;
}

.panther-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  padding-right: 8px;
}

.panther-37 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  width: 100%;
}

.panther-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  margin-bottom: 8px;
}

.panther-2 {
  font-weight: 500;
  margin-right: auto;
  max-width: 60%;
  word-break: break-word;
}

.panther-1 {
  -webkit-transition: color 0.1s ease-out;
  transition: color 0.1s ease-out;
  font-weight: 500;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #7fd3ff;
}

.panther-1:hover {
  color: #ace2ff;
}

.panther-1:focus {
  color: #ace2ff;
}

.panther-1:active,
.panther-1[data-active=true] {
  color: #11a5f3;
}

.panther-4 {
  font-size: 0.75rem;
  max-width: 20%;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.panther-3 {
  margin-right: 4px;
  color: #86a3c3;
}

.panther-7 {
  margin-right: auto;
}

.panther-6 {
  background-color: #141c2a;
  border-radius: 2px;
  padding-left: 6px;
  padding-right: 6px;
  padding-top: 4px;
  padding-bottom: 4px;
  font-weight: 700;
  font-size: 0.625rem;
  color: #29bac6;
}

.panther-36 {
  display: grid;
  grid-gap: 8px;
  grid-template-columns: 3fr 4fr 3fr;
}



.panther-14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: 8px;
}

.panther-8 {
  color: #bdbdbd;
  font-size: 0.563rem;
  font-weight: 500;
}

.panther-13 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  font-size: 0.875rem;
  font-weight: 500;
  opacity: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-height: 24px;
}

.panther-12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.panther-12>*:not(:last-child) {
  margin-right: 8px;
}

.panther-9 {
  max-width: 250px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.panther-11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 24px;
  height: 24px;
  background-color: #4f627c;
  border-color: #4f627c;
  border-radius: 99999px;
}

.panther-11:hover {
  background-color: #11a5f3;
  border-color: #11a5f3;
}

.panther-10 {
  display: inline-block;
  vertical-align: sub;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 12px;
  height: 12px;
  color: currentColor;
}

.panther-20 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.panther-20>*:not(:last-child) {
  margin-right: 8px;
}

.panther-17 {
  width: 12px;
  height: 12px;
  background-color: hsl(159, 29%, 39%);
  border-radius: 99999px;
}

.panther-18 {
  font-size: 0.75rem;
  font-weight: 400;
  line-height: 1.5;
}

.panther-35 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.panther-34 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: flex-end;
  justify-content: flex-end;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
}

.panther-34>*:not(:last-child) {
  margin-right: 8px;
}

.panther-25 {
  height: 18px;
}

.panther-24 {
  display: inline-block;
  vertical-align: sub;
  -webkit-animation: animation-0 2s linear infinite;
  animation: animation-0 2s linear infinite;
  width: 18px;
  height: 18px;
}

.panther-23 {
  stroke-dashoffset: 600;
  stroke-dasharray: 300;
  stroke-width: 13;
  stroke-miterlimit: 10;
  stroke-linecap: round;
  -webkit-animation: animation-1 1.6s cubic-bezier(0.4, 0.15, 0.6, 0.85) infinite;
  animation: animation-1 1.6s cubic-bezier(0.4, 0.15, 0.6, 0.85) infinite;
  stroke: currentColor;
  fill: transparent;
}

.panther-26 {
  border: none;
  opacity: 0.6;
  border-left: 0.0625rem solid;
  height: auto;
  margin-left: 0;
  margin-right: 0;
  border-color: inherit;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
}

.panther-27 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: default;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  min-width: 85px;
  text-align: center;
  font-weight: 700;
  border-radius: 40px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  font-size: 0.75rem;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 16px;
  padding-right: 16px;
  border: 1px solid;
  border-color: #ffa500;
  background-color: transparent;
}

.panther-33 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.panther-33>*:not(:last-child) {
  margin-right: 4px;
}

.panther-29 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 85px;
}

.panther-28 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: default;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  min-width: 85px;
  text-align: center;
  font-weight: 700;
  border-radius: 40px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  font-size: 0.75rem;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 16px;
  padding-right: 16px;
  background-color: #6e3078;
}

.panther-32 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.panther-31 {
  outline: none;
  line-height: 1;
  border-radius: 99999px;
  height: 24px;
  width: 24px;
  -webkit-transition: border-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,background-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: border-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,background-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  border: 1px solid;
  border-color: #3e516b;
  background-color: transparent;
  cursor: pointer;
  color: #f6f6f6;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.panther-31:hover {
  background-color: #465973;
  border-color: #465973;
}

.panther-31:focus {
  background-color: #465973;
  border-color: #93a6c0;
}

.panther-31:active,
.panther-31[data-active=true] {
  background-color: #253852;
  border-color: #253852;
}

.panther-30 {
  display: inline-block;
  vertical-align: sub;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 16px;
  height: 16px;
  color: currentColor;
}

<div>
  <header
    id="main-header"
  />
  <section
    class="panther-40"
  >
    <div
      class="panther-39"
    >
      <div
        class="panther-38"
      >
        <div
          class="panther-0"
        />
        <div
          class="panther-37"
        >
          <header
            class="panther-5"
          >
            <h4
              class="panther-2"
            >
              <a
                aria-label="Link to Alert"
                class="panther-1"
                href="/log-analysis/alerts/f67b8f04-5fac-404a-93a4-38db29f258ba/"
              >
                indexing
              </a>
            </h4>
            <div
              aria-label="Creation time for f67b8f04-5fac-404a-93a4-38db29f258ba"
              class="panther-4"
            >
              <span
                class="panther-3"
              >
                Created
              </span>
              2020-08-08 12:15 GMT
            </div>
          </header>
          <div
            class="panther-7"
          >
            <div
              class="panther-6"
            >
              POLICY FAIL
            </div>
          </div>
          <div
            class="panther-36"
          >
            <div
              class="panther-15"
            >
              <dl
                class="panther-14"
              >
                <dt
                  aria-labelledby="policyobject-object"
                  class="panther-8"
                >
                  Policy
                </dt>
                <dd
                  aria-labelledby="policyobject-object"
                  class="panther-13"
                >
                  <div
                    class="panther-12"
                  >
                    <span
                      class="panther-9"
                    >
                      a68babd7-7c1c-4dee-a33e-b8009e6d8017
                    </span>
                    <a
                      aria-label="Link to policy a68babd7-7c1c-4dee-a33e-b8009e6d8017"
                      href="/detections/policies/a68babd7-7c1c-4dee-a33e-b8009e6d8017/"
                    >
                      <div
                        class="panther-11"
                      >
                        <svg
                          class="panther-10"
                          role="presentation"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 4L10.59 5.41L16.17 11H4V13H16.17L10.59 18.59L12 20L20 12L12 4Z"
                          />
                        </svg>
                      </div>
                    </a>
                  </div>
                </dd>
              </dl>
            </div>
            <dl
              class="panther-14"
            >
              <dt
                aria-labelledby="resource-typesobject-object"
                class="panther-8"
              >
                Resource Types
              </dt>
              <dd
                aria-labelledby="resource-typesobject-object"
                class="panther-13"
              >
                <div
                  class="panther-20"
                >
                  <div
                    class="panther-12"
                  >
                    <span
                      class="panther-17"
                    />
                    <span
                      class="panther-18"
                    >
                      brand
                    </span>
                  </div>
                </div>
              </dd>
            </dl>
            <div
              class="panther-35"
            >
              <div
                class="panther-34"
              >
                <div
                  class="panther-25"
                >
                  <svg
                    aria-label="Loading..."
                    class="panther-24"
                    viewBox="0 0 150 150"
                  >
                    <circle
                      class="panther-23"
                      color="current"
                      cx="75"
                      cy="75"
                      r="60"
                    />
                  </svg>
                </div>
                <hr
                  aria-orientation="vertical"
                  class="panther-26"
                />
                <div
                  aria-atomic="true"
                  class="panther-27"
                  role="status"
                >
                  MEDIUM
                </div>
                <div
                  class="panther-33"
                >
                  <div
                    class="panther-29"
                  >
                    <div
                      aria-atomic="true"
                      class="panther-28"
                      role="status"
                    >
                      TRIAGED
                    </div>
                  </div>
                  <div
                    aria-controls="menu--1"
                    aria-haspopup="true"
                    class="panther-32"
                    data-reach-menu-button=""
                    id="menu-button--menu--1"
                    type="button"
                  >
                    <button
                      aria-label="Change Alert Status"
                      aria-pressed="false"
                      class="panther-31"
                      type="button"
                    >
                      <svg
                        class="panther-30"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M7 10L12 15L17 10H7Z"
                        />
                      </svg>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
  <footer
    id="footer"
  />
</div>
`;
